playbooks/kubernetes.md

```diff
@@ -71,8 +85,35 @@ To diagnose application failures, navigate to the Kubernetes dashboard and searc
 view on your application's resources, look for any Pods in an "Error" or "CrashLoopBackoff" state, then _Delete_
 these pods by either using the three-dots "actions" menu to the right of the pod's CPU / Memory graphs, or click
-through to the Pod view and use the "Delete" menu with the trash can at the top of the page.
+through to the Pod view and use the "Delete" menu with the trash can at the top of the page, or via the `kubectl` CLI: `kubectl delete {pod-name}`
 
 Alternatively, you can (in the application's git checkout, assuming you have the Hokusai CLI installed) run the
-command `hokusai [staging|production] refresh` but be aware this is a _hard_ restart of the application and may
-result in brief downtime as it will delete _all_ running pods simultaneously.
+command `hokusai [staging|production] refresh`
+
+#### When things go wrong (cluster level)
+
+Kubernetes cluster nodes may experience issues with their underlying system (too many open file handles, full disk, memory, etc) and require system-level debugging, or occasionally need to be re-provisioned.
+
+To inspect a node, first find its FQDN, which is the private DNS name assigned by AWS EC2 autoscaling.
+
+To view all nodes for a given cluster, run `kubectl get nodes`.  Another useful command, if you wish to see pods along with the node on which they are scheduled / running is `kubectl get pods -o wide`
+
+To ssh into an instance, first connect to the staging / production VPN then run `ssh -i ~/.artsyow.pem admin@{node-FQDN}`
+
+Nodes run services using [systemd](https://wiki.debian.org/systemd).  Use `systemctl status` to see all running services or `systemctl status docker.service` to inspect the docker daemon service.  Get logs using [`journalctl`](https://manpages.debian.org/stretch/systemd/journalctl.1.en.html) to view logs for systemd services, so for example to see recent logs for the docker service use `journalctl -u docker.service --follow`
+
+We pre-install useful packages for system level debugging in our base AMI for Kubernetes machines, so you have `htop` / `lsof` / `netstat` / `tcpdump` all available.  Another useful tool that aims to provide debugging for container environments is [`sysdig`](https://github.com/draios/sysdig/wiki/sysdig-user-guide) and its curses-interface [`csysdig`](https://github.com/draios/sysdig/wiki/Csysdig-Overview).  With it you can get stacktraces, inspect file allocations, network connections, etc while filtering on a specific process or container name.
+
+If a node is for any reason unable to schedule containers, serve traffic or unresponsive, it can safely be terminated - AWS autoscaling will replace the machine automatically with a new one and it will rejoin the cluster.  However, terminating a node that is running application pods may result in service disruption for that application if they are the _only_ pods that back that service.
+
+To evict all running pods from a node before terminating it, use the following procedure:
```

Comment:
Do we want to point developers to the [existing `rotate_cluster_nodes`](https://github.com/artsy/substance/blob/c7dee522551e764ff7703af3056af502a695e1cd/manage.py#L350-L398) procedure in artsy/substance? I'd figure that we to minimize human intervention whenever possible.